pycsw/wsgi.py

```diff
@@ -55,18 +30,53 @@
 import os
 import sys
 import six
+import re
+import ConfigParser
 from six.moves.urllib.parse import unquote
 
 from pycsw import server
 
 
 PYCSW_ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
+def dynUpdateConfigurationURLs(new_url):
+    '''
+    Update dynamically PyCSW configuration to match public IP:port of this server.
+
+    The PyCSW server can be deployed in a VM or a container in different environment (dev, prod...). Its IP:port is therefore not deterministic.
+    The values of 'serveraddress' in default.cfg can't be forecasted in advance.
+
+    This function is meant to be called the client emit a request.
+    The base URL requested by the client is retrieved and reinjected in PyCSW configuration.
+    '''
+    confFile = os.environ.get('PYCSW_CONFIG', '/etc/pycsw/default.cfg')
```

Comment:
Reading the config file from `/etc/pycsw/default.cfg` breaks existing behaviour of reading from `<pycsw_install_dir>/default.cfg`, apparently for no good reason. This is what is causing all the [Travis CI](https://travis-ci.org/geopython/pycsw/builds/169494574) failures. Please fix this in order to read from the expected location, whenever there is no `PYCSW_CONFIG` environment variable defined.